src/list/test.rs

```diff
@@ -22,6 +22,71 @@ mod iter {
     use super::*;
     use pretty_assertions::assert_eq;
 
+    #[test]
```

Comment:
These tests should probably be before `fn test_serde()`.  Note that this is the `iter` module.